nutils/solver.py

```diff
@@ -763,6 +763,8 @@ def _optimize(target, functional: evaluable.asarray, constrain, arguments, tol:
     dtype = _determine_dtype(target, (functional,), lhs0, constrain)
     mask, vmask = _invert(constrain, target)
     lhs, vlhs = _redict(lhs0, target, dtype)
+    if functional.shape != ():
+        raise ValueError('The objective function must be a scalar function of a single variable')
```

Comment:
Optimize can be used for multiple variables, so the second part of this message is false. A valid message could be simply "the objective function must be scalar valued".